agent/cache-types/streaming_health_services.go

```diff
@@ -30,13 +32,16 @@ func (c *StreamingHealthServices) Fetch(opts cache.FetchOptions, req cache.Reque
 	subscribeReq := stream.SubscribeRequest{
 		Topic:  stream.Topic_ServiceHealth,
 		Key:    reqReal.ServiceName,
+		Token:  reqReal.Token,
 		Index:  reqReal.MinQueryIndex,
 		Filter: reqReal.Filter,
-		TopicFilters: &stream.Filters{
-			Connect: reqReal.Connect,
-			Tags:    reqReal.ServiceTags,
-		},
 	}
+
+	// Switch the topic if Connect is enabled.
+	if c.Connect {
```

Comment:
per comment above this could just switch to the request:

```suggestion
	if reqReal.Connect {
```